### PR TITLE
Feature/pass iql string

### PR DESCRIPTION
**Pull request by p-louis**
Added option to pass iql_string as query parameter for objects

### CODE REVIEW

File: src/main/kotlin/com/linktime/plugin/jira/insightobjectgraph/impl/objects/ListObjectsEndpoint.kt [hunk 1]
```diff
@@ -40,8 +40,11 @@ constructor(private val listService: ListObjectsService) {
                        @QueryParam("schemaId") schemaId: Int?,
                        @QueryParam("attributes") attributes: Set<String>,
                        @QueryParam("resolve_attributes") resolveAttributes: Set<String>,
-                       @QueryParam("resolve_relations") resolveRelations: Boolean): Response {
-        val siteObjects = listService.getObjects(objectTypeName, schemaId, attributes, resolveAttributes, resolveRelations)
+                       @QueryParam("resolve_relations") resolveRelations: Boolean,
+                       @QueryParam("iql_string") iqlString: String): Response {
```
Comment (weickmanna) [inline src/main/kotlin/com/linktime/plugin/jira/insightobjectgraph/impl/objects/ListObjectsEndpoint.kt]: The iql_string parameter is optional. If I call the endpoint without the parameter, will it actually be bound here to the empty string automatically, or will it be null?

File: src/main/kotlin/com/linktime/plugin/jira/insightobjectgraph/impl/objects/ListObjectsServiceImpl.kt [hunk 2]
```diff
@@ -109,6 +109,31 @@ constructor(private val insightApiFacade: InsightApiFacade,
                 .let { ListObjectsService.InsightObjectListRootNode(it) }
     }
 
+    override fun getObjectsWithIql(objectTypeName: String,
```
Comment (weickmanna) [inline src/main/kotlin/com/linktime/plugin/jira/insightobjectgraph/impl/objects/ListObjectsServiceImpl.kt]: This method contains duplicated code with "getObjects". A separate method does not seem necessary if the iqlString was defined as nullable (as is done with schemaId).

**Review by weickmanna — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/linktime/plugin/jira/insightobjectgraph/impl/objects/ListObjectsEndpoint.kt (hunk 1 above)
Comment (p-louis) [reply src/main/kotlin/com/linktime/plugin/jira/insightobjectgraph/impl/objects/ListObjectsEndpoint.kt]: Updated it to nullable, it supposedly should have worked before though, as most of the QueryParameters are Optional but only 'schemaId' is nullable

File: src/main/kotlin/com/linktime/plugin/jira/insightobjectgraph/impl/objects/ListObjectsServiceImpl.kt (hunk 2 above)
Comment (p-louis) [reply src/main/kotlin/com/linktime/plugin/jira/insightobjectgraph/impl/objects/ListObjectsServiceImpl.kt]: Refactored to have only getObjects with the additional parameter

**Pull request merged**